Probability Sampling With Python

Import python libraries

In [1]:
import pandas as pd
import statistics   #pandas statistics module
import numpy as np
import random       #python method for choosing elements randomly from a set
from math import sqrt
from scipy.stats import sem 


Add dataset of or population data

In [2]:
#data set set including the age of 50 people

age = [23, 30, 21, 28, 25, 19, 35, 40, 27, 22, 
 31, 24, 29, 33, 38, 26, 24, 28, 32, 34, 
 36, 22, 25, 27, 41, 23, 32, 30, 29, 33, 
 35, 21, 28, 26, 24, 31, 38, 40, 29, 23, 
 21, 22, 25, 26, 39, 28, 30, 35, 34, 24
]

Calculate population mean of the age dataset

In [3]:
#calculating population mean of their ages

print(statistics.mean(age))                         #using pandas

population_mean = np.mean(age)                      #using numpy
print (population_mean)


28.92
28.92


Calculate sample mean

In [4]:
#get a random sample from the population(age) using Python's random.choices modules to return multiple random ages from the population

sample_size= len(age)
sample = random.choices(list(age), k=sample_size)
#print(sample)

#calculate sample mean
print(statistics.mean(sample))                     #using pandas

sample_mean = np.mean(sample)                      #using numpy
print (sample_mean)


29.08
29.08


Calculate the percentage standard error mean (SEM= S/square root of n) where;
S is the sample standard deviation
n is sample size

In [10]:
# Calculate the sample standard deviation
std_dev = np.std(age, ddof=1)  # ddof=1 to for sample std dev instead of population std dev, numpy

# Calculate the Standard Error of the Mean (SEM)
n = sample_size
sem = std_dev / np.sqrt(n)

# Calculate the difference between population mean and sample mean
#mean_difference = population_mean - sample_mean

# Calculate the percentage of SEM relative to the sample mean
sem_percentage = (sem / sample_mean) * 100

print(sem_percentage)

2.8217751871821277


Confidence Intervals with SEM to measure how well the sample represents the population

In [8]:
lower_limit = sample_mean - sem
upper_limit = sample_mean + sem

print(f"Confidence Interval: ({lower_limit}, {upper_limit})")

Confidence Interval: (28.259427775567435, 29.900572224432562)


therefore, the average age is probably between 28.25 and 29.90, and the sample is an accurate representation of the population since the population mean falls within this range